htlcswitch/switch.go

```diff
@@ -1628,15 +1641,16 @@ out:
 		// collect all the forwarding events since the last internal,
 		// and write them out to our log.
 		case <-s.cfg.FwdEventTicker.Ticks():
-			s.wg.Add(1)
-			go func() {
-				defer s.wg.Done()
-
-				if err := s.FlushForwardingEvents(); err != nil {
+			// The error of Go is ignored: if it is shutting down,
+			// the loop will terminate on the next iteration, in
+			// s.gm.Done case.
+			_ = s.gm.Go(background, func(ctx context.Context) {
```

Comment:
let `htlcForwarder` take a context and pass in a context in there from the goroutine which is starting it